python-scripts/get_kernel_keys.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017 Alexander McWhirter
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Tool for retrieving and displaying kernel keys from Xbox memory
+#
+# For more information about the Xbox kernel, see:
+# http://xboxdevwiki.net/Kernel
+
+from xbox import *
+import argparse
+import textwrap
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-d', action='store_true', dest='dump',
+                  help='dump security keys to a file (combine with -s)')
+parser.add_argument('-s', action='store_true', dest='sec',
+                  help='view security keys found in kernel memory space')
+parser.add_argument('-o', action='store_true', dest='other',
+                  help='view other keys found in kernel memory space')
+parser.add_argument('-x', action='store_true', dest='exec',
+                  help='view executable keys found in kernel memory space')
+args = parser.parse_args()
+
+eepKey = read(ke.XboxEEPROMKey(), 16, False)
+hddKey = read(ke.XboxHDKey(), 16, False)
```

Comment:
from xbox-dump: ` uint8_t* XboxCERTKey = &XboxHDKey[-16]; // XboxCERTKey is just infront of XboxHDKey`
Or translated to python: `certKey = read(ke.XboxHDKey() - 16, 16, False) # XboxCERTKey is just infront of XboxHDKey`

This key should be present, and be at that location for almost all kernels. We should also output it.
If possible, it should be warned that the CERT key might not be correct [as it's purely a guess where it is located in memory].

iirc it's one of the keys being passed to the kernel at startup, so it's actually an important key for emulation.